bigquery/google/cloud/bigquery/dataset.py

```diff
@@ -283,28 +293,38 @@ def from_string(cls, dataset_id, default_project=None):
             >>> DatasetReference.from_string('my-project-id.some_dataset')
             DatasetReference('my-project-id', 'some_dataset')
 
+            >>> DatasetReference.from_string('prefix:my-project-id.some_dataset')
```

Comment:
It is rare for non-Googlers to encounter such a project ID. I don't think it warrants a second code example in the docs.